File: Back/src/main/java/com/backend/model/KnowledgeSpace.java
package com.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.util.Set;

@Entity
public class KnowledgeSpace {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "name")
    private String name;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "subjectId")
    private Subject subjectId;

    @OneToMany(mappedBy = "knowledgeSpaceId",cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JsonIgnore
    private Set<Surmise> surmises;

    public KnowledgeSpace() {
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Subject getSubjectId() {
        return subjectId;
    }

    public void setSubjectId(Subject subjectId) {
        this.subjectId = subjectId;
    }

    public Set<Surmise> getSurmises() {
        return surmises;
    }

    public void setSurmises(Set<Surmise> surmises) {
        this.surmises = surmises;
    }
}

File: Front/Front/src/app/model/KnowledgeSpace.ts
import { Surmise } from "./Surmise";

export class KnowledgeSpace{

    id: number;
    name: string;
    subjectId: number;
    realSpace: boolean;
    surmises: Surmise[];
    constructor(data: any) {
      if (data) {
        this.id = data.id;
        this.name = data.name;
        this.subjectId = data.subjectId;
        this.realSpace = data.realSpace;
        this.surmises = data.surmises;
      }
  }}
  